src/web/client/utilities/commonUtil.ts

```diff
@@ -199,10 +202,24 @@ export function isPortalVersionV2(): boolean {
     return WebExtensionContext.currentSchemaVersion.toLowerCase() === portalSchemaVersion.V2;
 }
 
-export function getWorkSpaceName(websiteId : string) : string {
+export function getWorkSpaceName(websiteId: string): string {
     if (isPortalVersionV1()) {
         return `Site-v1-${websiteId}`;
     } else {
         return `Site-v2-${websiteId}`;
     }
 }
+
+// ENV_ID is the last part of the parameter value sent in the vscode URL from studio
+export function getEnvironmentIdFromUrl() {
+    return (WebExtensionContext.urlParametersMap.get(queryParameters.ENV_ID) as string).split("/")?.pop() as string;
+}
+
+export function getBackToStudioURL() {
+    const region = WebExtensionContext.urlParametersMap.get(queryParameters.REGION) as string;
+
+    return BACK_TO_STUDIO_URL_TEMPLATE
+        .replace("{environmentId}", getEnvironmentIdFromUrl())
+        .replace("{.region}", region.toLowerCase() === STUDIO_PROD_REGION ? "" : `.${WebExtensionContext.urlParametersMap.get(queryParameters.REGION) as string}`)
+        .replace("{webSiteId}", WebExtensionContext.urlParametersMap.get(queryParameters.WEBSITE_ID) as string);
```

Comment:
Is the expectation we always fallback to Homepage even though user navigates from other page.

Reply:
that is correct